### PR TITLE
Skip initialization in eigen problem when restarting

### LINKED ISSUE #27621: Skip initialization when restarting an eigenvalue calculation
### Motivation

It is allowed to use a restart file to set the initial condition for an eigenvalue problem. In this case, we should skip the initialization.

### Design

We did not consider restart steady-state eigenvalue calculation before. What we need to do is to add few checks with `_app.isRestarting()` to skip few initializations. I will submit a PR for this.

### Impact

Make the restart work as expected for an eigenvalue calculations. For example, if we restart from a checkpoint file generated by a previously converged solution, we should expect the calculation return immediately with the converged residual.

### PR DESCRIPTION
**Pull request by YaqiWang**
Closes #27621.

### CODE REVIEW

File: test/tests/problems/eigen_problem/initial_condition/ne_restart.i [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+# the minimum eigenvalue of this problem is 2*(PI/a)^2;
+# Its inverse is 0.5*(a/PI)^2 = 5.0660591821169. Here a is equal to 10.
+
+[Problem]
+  restart_file_base = ne_ic_out_cp/LATEST
+[]
+
+[Variables]
+  [./u]
```
Comment (lindsayad) [inline test/tests/problems/eigen_problem/initial_condition/ne_restart.i:21]: update to new syntax without `./` and `../`

File: test/tests/problems/eigen_problem/initial_condition/tests [hunk 2]
```diff
@@ -76,4 +76,15 @@
     requirement = "The system shall support eigenvalue calculations with an initial condition including the initial solution vector and the initial eigenvalue."
     slepc = true
   [../]
+
+  [./restart]
```
Comment (lindsayad) [inline test/tests/problems/eigen_problem/initial_condition/tests:80]: same. You can do the same for the rest of the file or leave it, let's just make the diff clean at least

**Review by lindsayad — CHANGES_REQUESTED** [review]

**Comment by YaqiWang** [comment]
Interesting. The failed test here shows a warning message at the end `WARNING! There are options you set that were not used!`.

**Comment by YaqiWang** [comment]
@lindsayad more comments? The failures do not seem relevant.

**Review by lindsayad — APPROVED** [review]

**Comment by lindsayad** [comment]
indeed, failures are not related

**Comment by YaqiWang** [comment]
Thanks @lindsayad!

**Comment by loganharbour** [comment]
Broke next https://civet.inl.gov/job/2234270/

**Comment by loganharbour** [comment]
Can someone address this please

**Comment by YaqiWang** [comment]
That is strange: why p3 and p7 tests pass and p5 does not? I cannot reproduce it on my mac...

**Pull request merged**